### PR TITLE
[New Rules] cap_setuid/cap_setgid privesc

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This PR proposes two new rules:
1. Potential Privilege Escalation via Python cap_setuid
2. Setcap setuid/setgid Capability Set

### Potential Privilege Escalation via Python cap_setuid
This detection rule monitors for the execution of a system command with setuid or setgid capabilities via Python, followed by a uid or guid change to the root user. This sequence of events may indicate successful privilege escalation. Setuid (Set User ID) and setgid (Set Group ID) are Unix-like OS features that enable processes to run with elevated privileges, based on the file owner or group. Threat actors can exploit these attributes to escalate privileges to the privileges that are set on the binary that is being executed. 

#### Detection
**0 FPs in telemetry last 365 days, only TPs in test stack**
```
sequence by host.id, process.entity_id with maxspan=1s
  [process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
   process.args : "import os;os.set?id(0);os.system(*)" and process.args : "*python*" and user.id != "0"]
  [process where host.os.type == "linux" and event.action in ("uid_change", "gid_change") and event.type == "change" and 
   (user.id == "0" or group.id == "0")]
```

image

### Setcap setuid/setgid Capability Set
This rule monitors for the addition of the cap_setuid+ep or cap_setgid+ep capabilities via setcap. Setuid (Set User ID) and setgid (Set Group ID) are Unix-like OS features that enable processes to run with elevated privileges, based on the file owner or group. Threat actors can exploit these attributes to achieve persistence by creating malicious binaries, allowing them to maintain control over a compromised system with elevated permissions.

#### Detection
**0 FPs in telemetry last 365 days, only TPs in test stack**
```
process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
process.name == "setcap" and process.args : "cap_set?id+ep"
```

image

### CODE REVIEW

**Review by imays11 — APPROVED** [review]

File: rules/linux/persistence_setuid_setgid_capability_set.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+name = "Setcap setuid/setgid Capability Set"
+risk_score = 47
+rule_id = "f5c005d3-4e17-48b0-9cd7-444d48857f97"
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Defend"]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
+process.name == "setcap" and process.args : "cap_set?id+ep"
```
Comment (w0rk3r) [inline rules/linux/persistence_setuid_setgid_capability_set.toml:30]: >cap_set?id+ep
🧙🏼 ++

File: rules/linux/persistence_setuid_setgid_capability_set.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+'''
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+
+[rule.threat.tactic]
```
Comment (w0rk3r) [inline rules/linux/persistence_setuid_setgid_capability_set.toml:36]: We can use this one here, I think

https://attack.mitre.org/techniques/T1548/001/

File: rules/linux/privilege_escalation_suspicious_cap_setuid_python_execution.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/09/05"
+
+[rule]
+author = ["Elastic"]
+description = """
+This detection rule monitors for the execution of a system command with setuid or setgid capabilities via Python, 
+followed by a uid or guid change to the root user. This sequence of events may indicate successful privilege escalation.
```
Comment (w0rk3r) [inline rules/linux/privilege_escalation_suspicious_cap_setuid_python_execution.toml:13]:
```suggestion
followed by a uid or gid change to the root user. This sequence of events may indicate successful privilege escalation.
```
Idk, but gid seems to be a better fit in the context

File: rules/linux/privilege_escalation_suspicious_cap_setuid_python_execution.toml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+[[rule.threat.technique]]
+id = "T1068"
+name = "Exploitation for Privilege Escalation"
+reference = "https://attack.mitre.org/techniques/T1068/"
+
+[rule.threat.tactic]
+id = "TA0004"
+name = "Privilege Escalation"
+reference = "https://attack.mitre.org/tactics/TA0004/"
```
Comment (w0rk3r) [inline rules/linux/privilege_escalation_suspicious_cap_setuid_python_execution.toml:47]: We can also add this one:

https://attack.mitre.org/techniques/T1548/001/

**Review by w0rk3r — APPROVED** [review]
🧙🏼

File: rules/linux/privilege_escalation_suspicious_cap_setuid_python_execution.toml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+[[rule.threat.technique.subtechnique]]
+id = "T1548.001"
+name = "Setuid and Setgid"
+reference = "https://attack.mitre.org/techniques/T1548/001/"
+
+[rule.threat.tactic]
+id = "TA0004"
+name = "Privilege Escalation"
+reference = "https://attack.mitre.org/tactics/TA0004/"
```
Comment (Aegrah) [inline rules/linux/privilege_escalation_suspicious_cap_setuid_python_execution.toml:65]:
```suggestion
reference = "https://attack.mitre.org/tactics/TA0004/"


```

**Pull request merged**